File: app/src/main/java/com/jssg/servicemanagersystem/base/entity/BaseHttpResult.java
package com.jssg.servicemanagersystem.base.entity;

import android.text.TextUtils;

import java.io.Serializable;

public class BaseHttpResult<T> implements Serializable {

    public int code;
    public String message;
    public String status;
    public T data;

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public String getMessage() {
        if (!TextUtils.isEmpty(message)) return message;
        else if (!TextUtils.isEmpty(status)) return status;
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }


    public boolean isSuccess(){
        return code==0;
    }



}

File: app/src/main/java/com/jssg/servicemanagersystem/base/entity/BaseHttpResult.kt
package com.jssg.servicemanagersystem.base.entity

import java.io.Serializable

class BaseHttpResult<T> : Serializable {
    @JvmField
    var code = 0
    var msg: String? = null
    @JvmField
    var data: T? = null

    var rows: T? = null

    fun setData(data: T) {
        this.data = data
    }

    val isSuccess: Boolean
        get() = code == 0 || code == 200
}